### DISCUSSION TITLE
Do I need a physical controller for this to interface with?

### DISCUSSION THREAD

**Discussion by chadhobson** [tagyoureit/nodejs-poolController#462] [General]
So I'm changing out all of my outdated pool equipment and switching to Pentair. So far I have the following ready to be installed but still in the box, so I haven't poked around any of it...

- Pentair IntelliCom II Interface Adapter
- Pentair IntelliChlor PC100 Power Center
- Pentair IntelliChlor IC40 Chlorine Generator Cell Kit
- Pentair IntelliFlo i1 Variable Speed Pump 

1. Do I still need to buy an actual controller from Pentair or does this completely replace that?
2. I **think** the pump and SWG both have an RS-485 connection. If so, how do you connect two cables to one RS-485 adapter?
3. My preference is to run cabling from inside the house where my server rack is, out to the pool equipment pad. Any issues with that?

Thanks!

**Comment by rstrouse** [comment]
Answers
1. No you do not need any Pentair controller for this to work.  It will not interface with your IntelliCom II interface adapter but I don't see anything on that list that I would hook that to.
2. Yes the pump and the chlorinator have an RS485 interface.  RS485 is a bus architecture meaning that all like wires are joined together on the same connection.  So you simply connect both green and both yellow wires for your pump and chlorinator into the same RS485 adapter.
3. No issues with running ethernet or RS485 to your server rack.  The results will be the same.  The only difference is where the equipment will be located.

**Reply by chadhobson** [reply]
On #2...

I'll be running the controller on a VM on a Proxmox server in my rack. So can I run a single wire out from there to the equipment pad and mate up all the RS-485 wires at the pad or do I need to run two wires? From what you said, it sounds like I can get away with one wire but I just want to be sure.

I also thought of a couple other questions so bear with me...

4. Can I pull water temperature from the SWG or the pump or do I need to add a sensor?
5. I noticed in the Home Assistant issue someone posted that someone had their entire water chemistry showing in the dashboard. Is that the Intellichem providing that info or something else?

**Reply by rstrouse** [reply]
Answers (continued)

4. You cannot get temperature from the SWG or pump.  You need to add a sensor.  Fortunately, and unfortunately, we have written a ton of information as well as management applications to connect water temp sensors.  Look at the DIY Nixie wiki to see how standard water temp sensors are easily integrated. 
5. IntelliChem is one options but frankly you will have much better control and chemistry management with REM Chem.

I have included some light reading for your perusal.  The suggested equipment is just a suggestion and by no means required.  If you have any questions let us know.
https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller
https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-REM-Chem-Controller
https://github.com/rstrouse/relayEquipmentManager/wiki/Temperature-Sensors

**Reply by chadhobson** [reply]
Thank you for the info @rstrouse! I really appreciate it!

**Reply by chadhobson** [reply]
So I've read all of that now. Thanks again.

The Nixie article confirms my thought that the pool pump is always powered on, so no relay needed. But what about the chlorinator? I was hoping I could just run a single cable from my basement out to the pad and tie into both RS-485 connections and that would be enough to turn the chlorinator off/on and set pump speeds. But if I need a relay, now we're talking about a Pi in a box with a relay board on the pad. So do I need a relay?

Next year I'm ditching the current fiber lights and going LED, so I'll no doubt want to tie that in to the system which means I would need a relay at that point. So maybe I go the "pi in a box" route and future-proof myself a little?

**Reply by rstrouse** [reply]
Adding a relay for the filter and attaching the chlorinator to it is the safest bet.  The issue with chlorinators is that the flow switches go bad.  Unfortunately, when that happens it can generate chlorine gas in the pipes and expel that to an unexpected swimmer.  However, there is another danger.  The act of electrolysis generates hydrogen gas which is explosive.   If there is no flow and the cell is engaged this gas will accumulate in the plumbing and can explode.   Do a little perusing about exploding chlorinators.

https://lmddgtfy.net/?q=Exploding%20Chlorinator

**Reply by chadhobson** [reply]
I'm now ordering all the parts for the single body dream system.

On the 3HP relay... is there a DIN-mounted option? How is everyone mounting these in the box if there isn't?

In fact, is there a pic collection of folk's builds that I could reference?

**Reply by tagyoureit** [reply]
@rstrouse has a picture [in the wiki](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#keep-cool-and-be-a-fan).  

The DIN mounted 3hp relays are very expensive and we don't recommend putting them in the same box with the low voltage components.  You should put them in an adjacent box (spare timer box or other outdoor NEMA rated box).

**Reply by chadhobson** [reply]
Makes sense. Since I don't have the relay in hand, I have no idea how big/small it is. Are you suggesting something like a two gang outdoor junction box mounted below the larger low voltage box? Or is two gang not big enough?

**Reply by chadhobson** [reply]
Or maybe something like this from [Lowes](https://www.lowes.com/pd/Kraloy-6-in-x-6-in-x-4-in-PVC-Conduit-Junction-Box/5001212789)? Or better yet... [this one](https://www.sprinklerwarehouse.com/k-rain-24v-coil-pump-start-relay-1522) has a relay inside a box for a decent price.

**Reply by tagyoureit** [reply]
Just depends on how many relays you need.  This box is just a bit larger [Orbit enclosure](https://www.lowes.com/pd/Orbit-Gray-Silver-Timer-Cabinet/1128857) if you need more room.

**Reply by rstrouse** [reply]
Actually if there is a breaker box at your pad it is a perfect place for them.

**Reply by chadhobson** [reply]
Breaker box is by the house and runs out to the pad. I only need the one relay as I only have one pump, so I think I'm going to go for that second link above. Mount it below the main low voltage box. I think I'm finally getting my mind around all the hardware which is great because I'm a software guy!

Thanks to you all for the continued help. I'm sure I'll have more questions but I feel pretty good about everything except what I'm paying for all this stuff. Oof.

**Reply by rstrouse** [reply]
I hear ya.  I too am a software guy.  Hardware is a necessary and unfortunate evil.  It is shocking how much more expensive everything is.

**Reply by chadhobson** [reply]
I just paid $180 for a board-only 8GB Raspberry Pi. How insane is that?!?

**Reply by rstrouse** [reply]
It is crazy considering there was a time where that could be had for 49 bones.

**Reply by chadhobson** [reply]
So there's a CR2450 coin battery on the automation hat so it can "keep time indefinitely". 

1. I see it's empty in the Nixie Wiki link so I guess its not needed?
2. If it's needed, how is the clock time set?

**Reply by rstrouse** [reply]
njsPC will use ntp services to keep the time from the pi os.  You need to set your time zone and let njsPC know your latitude and longitude under the general tab.  It can then calculate sunrise and sunset for use in your schedules.

**Reply by chadhobson** [reply]
Dangit. So I need to run out and get a battery before I stack these things too deep?

**Reply by rstrouse** [reply]
If you have your pi connected so that it can see the Internet you do not need to run out and get a battery.  Sorry I wasn't too clear and sometimes I don't realize that folks don't know that there is an array of servers floating out there in the clouds that keep time.  Installing the battery isn't a terrible idea as it can be used to make sure the pi doesn't loose time if it doesn't see the Internet or your router (which in most instances has its own time server in it).

**Reply by chadhobson** [reply]
Yes, I'm running it on wifi for now and it's going to be at the limits of my range until I get a hard line buried out to it, so I'm not entirely sure how well connected its going to be yet.

**Reply by chadhobson** [reply]
New pump is in and semi-working. Won't hold prime down in the sub-1K RPM range but thats fine. We'll look at that later.

The [pump relay](https://www.sprinklerwarehouse.com/k-rain-24v-coil-pump-start-relay-1522) arrived today and I noticed the website said 24V but now that I have it in my hand it says 24 VAC. But I'm running the SDR-240-48 which is 24VDC.

Did I buy the wrong thing or can I use this for a pump relay?

**Reply by rstrouse** [reply]
You can use that for the pump relay however you cannot trip it with the SDR-240-48.  Typically I have used these with 24vdc
https://www.amazon.com/Pentair-520106-Assembly-Replacement-Control/dp/B002NJM1GC/ref=psdc_1272941011_t1_B0039Q9UWG

Are you planning on implementing any motorized actuators?  If you are then you will need a 24vac transformer anyway and if you have used the Sequent MEGA-BAS you can power it with this transformer as well.
https://www.supplyhouse.com/Functional-Devices-TR100VA002-Transformer-w-Circuit-Breaker-Foot-Dual-Threaded-Hub-Mount-100VA-120-to-24-Vac?utm_source=google_ad&utm_medium=Shopping_ntm&utm_campaign=Shopping_NTM_HPLTV&gclid=EAIaIQobChMIh-eql5Sa-AIV5R6tBh0WzwbKEAQYCiABEgKHMfD_BwE
